extensions/picocli/runtime/src/main/java/io/quarkus/picocli/runtime/PicocliCommandLineProducer.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.picocli.runtime;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.literal.NamedLiteral;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.quarkus.arc.DefaultBean;
+import io.quarkus.picocli.runtime.annotations.TopCommand;
+import picocli.CommandLine;
+
+@ApplicationScoped
+public class PicocliCommandLineProducer {
+
+    private final Instance<Object> topCommand;
+    private final PicocliConfiguration picocliConfiguration;
+
+    @Inject
+    public PicocliCommandLineProducer(@TopCommand Instance<Object> topCommand, PicocliConfiguration picocliConfiguration) {
+        this.topCommand = topCommand;
+        this.picocliConfiguration = picocliConfiguration;
+    }
+
+    private Class<?> classForName(String name) {
+        try {
+            return Class.forName(name);
+        } catch (ClassNotFoundException ex) {
+            throw new IllegalArgumentException(ex);
+        }
+    }
+
+    @Produces
+    @DefaultBean
```

Comment:
Are users expected to be able to provide their own `CommandLine.IFactory`? If not, then there is no need to have this annotation.

Reply:
I would say that normally, no, users would not provide their own factory. Also,  I can see no issue with the factory implementation in this extension.

However, why take away the possibility for users to provide their own factory?
There may be some use case that we haven't thought of and then we will be happy if the `@DefaultBean` annotation is there so it is easy for users to provide their own.

Reply:
I'm not opposed to it, just asking 😉

Reply:
Won't users be able to overwrite this anyway with Alternative + Priority(1)?

Reply:
Yes they can do that if needed.

Reply:
> Won't users be able to overwrite this anyway with Alternative + Priority(1)?

I see, that is good to know. In that case the difference is a bit academic and I am fine either way. 😉